### DISCUSSION TITLE
API documentation template a la Slate

### DISCUSSION THREAD

**Discussion by jaredmoody** [bridgetownrb/bridgetown#497] [Ideas]
At work we currently use [Slate](https://github.com/slatedocs/slate) to generate our API documentation. Slate is based on middleman, but IIRC are looking to remove ruby and move to a node-only implementation.

Generating static API docs seems _right_ in Bridgetown's wheelhouse - does anyone know of an existing project using Bridgetown that's similar to Slate, or might someone be interested in collaborating to create one?

**Comment by jaredcwhite** [comment]
@jaredmoody While you certainly _can_ use Bridgetown for documentation, it's not especially suited to the sort of really in-depth docs you need for APIs. I'd certainly be interested in collaborating on a trial run of it though!

**Comment by jaredmoody** [comment]
Is there some reason Bridgetown is less suited toward API docs than middleman, which Slate uses?

**Reply by jaredcwhite** [reply]
No, not that I know of. I'm sure it can be done…just a matter of who has the time or inclination to give building that extra layer a try.

**Comment by matoni109** [comment]
@jaredcwhite I'm looking at trying to do this with [spectaql](https://github.com/anvilco/spectaql) for our graphql docs.

This library generates the docs into a folder with below structure via a yarn script.

* what wold be the best way to integrate the below html / css / js / into the bridgetown build ? 

```markdown
  -- build
    -- images
      -- logo.svg
    -- javascripts
      -- spectaql.js
    -- stylesheets
      -- spectraql.css
    index.html
```

Would you be interested in exploring this ?
